packages/frontend/src/hooks/useTransaction.ts

```diff
@@ -0,0 +1,238 @@
+import { useState, useEffect } from 'react'
+import { getNetworkWaitConfirmations } from 'src/utils/networks'
+import {
+  fetchTransferFromL1Completeds,
+  fetchWithdrawalBondedsByTransferId,
+  getTransferSentDetailsFromLogs,
+  L1Transfer,
+  getLastLog,
+  formatLogArgs,
+  networkIdToSlug,
+} from 'src/utils'
+import { TransactionResponse, TransactionReceipt } from '@ethersproject/providers'
+import { formatEther } from '@ethersproject/units'
+import { getProviderByNetworkName } from 'src/utils/getProvider'
+import { Interface, Result } from '@ethersproject/abi'
+import { getExplorerTxUrl } from 'src/utils/getExplorerUrl'
+import { hopBridgeTokenAbi, l1BridgeAbi, l2AmmWrapperAbi } from '@hop-protocol/core/abi'
+
+export const methodToSigHashes = {
+  // L1_Bridge
+  sendToL2: '0xdeace8f5', // L1 -> L2
+  // L2_Bridge
+  swapAndSend: '0xeea0d7b2', // L2 -> L2 / L2 -> L1
+  // L1_Bridge / L2_Bridge
+  withdraw: '0x0f7aadb7', // L2 -> L2 / L2 -> L1 (bonder offline)
+}
+
+export const sigHashes = {
+  '0x095ea7b3': 'approve',
+  // L1_Bridge
+  '0xdeace8f5': 'sendToL2', // L1 -> L2
+  // L2_Bridge
+  '0xeea0d7b2': 'swapAndSend', // L2 -> L2 / L2 -> L1
+  // L1_Bridge / L2_Bridge
+  '0x0f7aadb7': 'withdraw', // L2 -> L2 / L2 -> L1 (bonder offline)
+}
+
+enum TxType {
+  approve = 'approve',
+  sendL1ToL2 = 'L1 to L2',
+  sendL2ToL2 = 'L2 to L2',
+  sendL2ToL1 = 'L2 to L1',
+  convert = 'convert',
+  wrap = 'wrap',
+  unwrap = 'unwrap',
+  addLiquidity = 'addLiquidity',
+  removeLiquidity = 'removeLiquidity',
+  stake = 'stake',
+  unstake = 'unstake',
+}
+
+interface TxObj {
+  type?: TxType
+  response: TransactionResponse
+  receipt: TransactionReceipt
+  gasCost: string
+  datetime?: string
+  methodName?: string
+  tokenSymbol?: string
+  params?: any[]
+  eventValues?: any
+  destNetworkName?: string
+  destTxHash?: string
+  destExplorerLink?: string
+}
+
+const useTransaction = (txHash?: string, slug?: string) => {
```

Comment:
another iteration of this hook is what i imagine would be a nice complement to the `Transaction` model